homeassistant/components/sma/__init__.py

```diff
@@ -57,7 +58,18 @@ async def _parse_legacy_options(entry: ConfigEntry, sensor_def: pysma.Sensors) -
     # Parsing of sensors configuration
     config_sensors = entry.data.get(CONF_SENSORS)
     if not config_sensors:
-        return
+        return []
+
+    # Support import of legacy config that should have been removed from 0.99, but was still functional
+    # See also #25880 and #26306. Functional support was dropped in #48003
+    if isinstance(config_sensors, dict):
+        config_sensors_list = []
+
+        for name, attr in config_sensors.items():
+            config_sensors_list.append(name)
+            config_sensors_list.extend(attr)
```

Comment:
Why are we adding both the key and the value to the list. In the test the values are lists. What is the intention here?